### ISSUE TITLE
[Bug]: Dock icon appears when using Always hidden feature.

### ISSUE THREAD

**Issue by illusian** [jordanbaird/Ice#397]
### Check Existing Issues

- [X] I have checked existing issues, and this issue is not a duplicate

### Description

The Ice dock icon appears when clicking the icon icon to show hidden icons when always hide enabled in advanced settings. This issue won't reproduce when "always hide" is disabled.
I tried with a fresh install, but the same issue persists.

### Steps to Reproduce

1. Settings.
2. Enable "always-hidden" section.
3. Click the Ice icon to show the hidden section. (Not always hidden section)

### Ice Version

0.11.9

### macOS Version

15.0.1

**Comment by jordanbaird** [comment]
Could you take a screen recording of this, with your entire screen? I think I know what's happening, I just want to verify.

**Comment by illusian** [comment]
https://github.com/user-attachments/assets/baa3bfc8-0c0b-45dc-85b7-7d539dbe9ef3

@jordanbaird Here you are.

**Comment by jordanbaird** [comment]
Oh, interesting. That should only be happening if the menus on the left are long enough to overlap with the shown icons. I can't reproduce it, but I'll try some other configurations when I can. You can disable the "Hide application menus when showing menu bar items" advanced setting for now to hopefully prevent this.

**Comment by illusian** [comment]
Yes, it won't happen when I disable the "Hide application menus when showing menu bar items" option.

**Comment by RottenHC** [comment]
Also experiencing this. Thank you for the temporary fix.

**Comment by daniellivingston** [comment]
+1 for macOS Tahoe 26.0. Can confirm that untoggling "Advanced -> Hide application menus when showing menu bar items" resolves the issue.

**Comment by milosdjakovic** [comment]
I also confirm this bug on macOS 26.0 (25A354). Turning off "Hide application menus when showing menu bar items" under "Advanced" settings resolves the issue.
